### ISSUE TITLE
lc0-tf -- training data is broken

### ISSUE THREAD

**Issue by mooskagh** [glinscott/leela-chess#471]
It's reported, that training data generated with lc0-tf with non-zero temperature, makes policy/reg loss equal to NaN while attempting training.

Here is example training set.
https://www.dropbox.com/s/vwvl03o9zkjgb3b/data-odd.tar?dl=0

Need to investigate what's wrong with that data and fix the cause.

I had a suspicion that the final state was also included into the training data, but it turned out not to be the case.

**Comment by blin00** [comment]
more specifics:

I generated that data with command
`./lc0 selfplay --training --noise --temperature=100 --visits=200 --weights=id197`
(temperature is 100x, so actual temperature is 1.0)
and gzipped the results myself afterwards

and then used this training config
```
%YAML 1.2
---
name: 'odd'
gpu: 0

dataset:
    num_chunks: 64
    train_ratio: 0.90
    input: '/path/to/data/'

training:
    batch_size: 32
    total_steps: 2000
    shuffle_size: 102400
    lr_values:
        - 0.001
        - 0.0001
    lr_boundaries:
        - 1000
    policy_loss_weight: 1.0
    value_loss_weight: 1.0
    path: '/path/to/output/'

model:
    filters: 64
    residual_blocks: 4
...
```
dropping the temperature to ~0.2 (`--temperature=20`) makes training not instantly blow up (but it's possible the data is still weird/wrong)

**Comment by mooskagh** [comment]
I found bad data, I write NaN into training data indeed. Debugging.
